kfp_tests/sample_flows/complex_flow_1.py

```diff
@@ -0,0 +1,135 @@
+        print(" BR1 -> BR11 and BR12 ")
+        print("\n\n_______________________________________")
+        self.next(self.br11, self.br12)
+
+    @step
+    def br2(self):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR2 -> BR21 and BR22 ")
+        print("___________________________________________")
+        self.next(self.br21, self.br22)
+
+    @step
+    def br11(self):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR11 -> JOIN1")
+        print("\n\n_______________________________________")
+        self.next(self.join1)
+
+    @step
+    def br12(self):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR 12 -> JOIN1 ")
+        print("___________________________________________")
+        self.next(self.join1)
+
+    @step
+    def br21(self):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR21 -> JOIN2")
+        print("\n\n_______________________________________")
+        self.next(self.join2)
+
+    @step
+    def br22(self):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR22 -> JOIN2 ")
+        print("___________________________________________")
+        self.next(self.join2)
+
+    @step
+    def join1(self, br1_inp):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR11 & BR12 -> JOIN1 ")
+        self.var1 = 200
+        print("Inside join1; var = ", self.var1)
+        print("___________________________________________")
+        self.next(self.join3)
+
+    @step
+    def join2(self, br2_inp):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" BR21 & BR22 -> JOIN2 ")
+        self.var2 = 100
+        print("Inside join2; var = ", self.var2)
+        print("___________________________________________")
+        self.next(self.join3)
+
+    @step
+    def join3(self, br_inp):
+        """
+        A step for metaflow to introduce itself.
+
+        """
+        print("\n\n_______________________________________")
+        print(" JOIN1 & JOIN2 -> JOIN3")
+        print("Verifying all input paths are correct by accessing same variable from"
+              "both branches of the join...")
+        print("JOIN1 var = ", br_inp.join1.var1)
+        print("JOIN2 var = ", br_inp.join2.var2)
+        print("___________________________________________")
```

Comment:
The flows should have assertions to ensure that it ran correctly.  The text says "Verifying all input paths are correct" but won't fail if they aren't.